pkg/stanza/operator/parser/json/config.go

```diff
@@ -32,6 +32,38 @@ func NewConfigWithID(operatorID string) *Config {
 // Config is the configuration of a JSON parser operator.
 type Config struct {
 	helper.ParserConfig `mapstructure:",squash"`
+
+	*JsoniterConfig `mapstructure:"jsoniter_config,omitempty"`
+}
+
+type JsoniterConfig struct {
+	IndentionStep                 int    `mapstructure:"indention_step"`
```

Comment:
Do we want to expose all these options to the users?

Reply:
I agree, my second thought on this is at least at this moment we should only expose options we needed, to not bind to the json iterator so tight.

Reply:
currently `json_parser` uses `jsoniter.ConfigFastest` which has 2 more options enabled

```go
var ConfigFastest = [Config](https://pkg.go.dev/github.com/json-iterator/go#Config){
	EscapeHTML:                    [false](https://pkg.go.dev/builtin#false),
	MarshalFloatWith6Digits:       [true](https://pkg.go.dev/builtin#true),
	ObjectFieldMustBeSimpleString: [true](https://pkg.go.dev/builtin#true),
}.Froze()
```

to add the `use_number` config, it looks like we have to create a new `jsoniter.Config` object probably to have the same config so users are not surprised, I think it makes sense to allow user to be able to turn off `MarshalFloatWith6Digits` as well. any thoughts or comments on this?

Reply:
I agree we should only expose what is needed, and we should consider that perhaps some day we need to switch json parsing libraries.

Reply:
I was wrong about `MarshalFloatWith6Digits`, it doesn't affect unmarshal, only marshal, which is not related to the parser. so for now I only added config for `UseNumber`.